File: Data Structures & Algorithms/max-water-container/submission-1.go
func maxArea(nums []int) int {
    res := 0
    l,r  := 0, len(nums)-1
    
    for l < r {
        if nums[l] > nums[r] {
            // fmt.Println(nums[r] * (r-l))
            if (nums[r] * (r-l)) > res {
                res = nums[r] * (r-l)
            }
            r--
        } else  {
            // fmt.Println(nums[l] * (r - l))
            if (nums[l] * (r - l)) > res {
                res = nums[l] * (r - l)
            }
            l++
        }
    }

    return res
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @returns {number[][]}
     */
    combinationSum(nums, target) {
        let res = []
        this.dfs(nums, 0, target, [], res)
        return res
    }


    dfs(nums, index,target, subset, res) {
        if (target === 0) {
            res.push([...subset])
        } else if (target < 0 || index >= nums.length) {
            return
        } else {
            subset.push(nums[index])
            this.dfs(nums, index, target - nums[index], subset, res)
            subset.pop()
            this.dfs(nums, index+1, target, subset, res)
        }
    }
}
